metronome/hotstuff/service/src/io/iohk/metronome/hotstuff/service/messages/HotStuffMessage.scala

```diff
@@ -0,0 +1,23 @@
+package io.iohk.metronome.hotstuff.service.messages
+
+import io.iohk.metronome.hotstuff
+import io.iohk.metronome.hotstuff.consensus.basic.Agreement
+
+/** Messages which are generic to any HotStuff BFT agreement. */
+sealed trait HotStuffMessage[A <: Agreement]
+
+object HotStuffMessage {
+
+  /** Messages which are part of the basic HotStuff BFT algorithm itself. */
+  case class ConsensusMessage[A <: Agreement](
+      message: hotstuff.consensus.basic.Message[A]
```

Comment:
I think it might be more intuitive to browse the project structure if this `Message` also lived in its own `messages` sub-package, symmetric to other modules

Reply:
You mean inside `consensus.basic.messages`? Yeah maybe, it could also perhaps be called `ProtocolMessage` to help recognise what it is. At the time I thought a flat package is easy enough to understand, everything is just about the `basic` protocol. 

I'd leave it there for now so there's less churn on this chain of PRs, but we can move it around if it makes it clearer.